### PR TITLE
CP-31450: Add domid to Datapath.attach

### PR DESCRIPTION
**Pull request by BenSimsCitrix**
We now honour the existing API of Datapath.attach by
passing the domid of vm that the vbd is being attached to.

This will enable a qemu process per vm.

Signed-off-by: ben sims <ben.sims@citrix.com>

### CODE REVIEW

File: main.ml [hunk 1]
```diff
@@ -528,7 +528,7 @@ let vdi_of_volume x =
     persistent = true;
   }
 
-let choose_datapath ?(persistent = true) response =
```
Comment (edwintorok) [inline main.ml]: Why the empty default? It might be better to be explicit about the domain always.

Comment (BenSimsCitrix) [reply main.ml]: fixed

**Comment by BenSimsCitrix** [comment]
This has passed

Storage BST Job ID 110188
DiffHostCrossPoolSXM Job ID 2594732 (Old version Lima)
DiffHostCrossPoolSXM Job ID 2594733 (Old version Dundee)

Latest changes have also built here.

https://ratchet.do.citrite.net/job/xenserver-build/job/private%252Fbensi%252FCA-312665/

**Comment by BenSimsCitrix** [comment]
This is ready for merge unless there are more review issues

**Pull request merged**